src/mono/wasm/runtime/jiterpreter-jit-call.ts

```diff
@@ -269,8 +269,9 @@ export function mono_jiterp_do_jit_call_indirect (
         }
     };
 
-    let failed = !getIsWasmEhSupported();
-    if (!failed) {
+    let failed = false;
+    const enabled = !runtimeHelpers.storeMemorySnapshotPending && getIsWasmEhSupported();
```

Comment:
This code path will result in mono_jiterp_do_jit_call_indirect running for every JIT call, which may not be what we want? 

Is it the case that once the snapshot is taken, storeMemorySnapshotPending will become false and then the next call to this will run the initialization logic? Or is it possible that it could stay true forever on the first run?

Reply:
To be clear, the problem is that `mono_jiterp_do_jit_call_indirect` is designed to run exactly once ever, and it's not fast. But by setting both `enabled` and `failed` to false, we ensure that it will keep getting called.

If we were to `cwraps.mono_jiterp_update_jit_call_dispatcher(0);` it would force the use of the C++ implementation instead, which is slower than indirect_js but faster than this.

Reply:
I only want to temporarily disable it until after snapshot is taken. It should be able to execute (un-optimized) before that.

Reply:
Now I see that it's only one function pointer. I will try to solve it.

Reply:
Solved, thanks for this.